proxy.go

```diff
@@ -42,12 +44,15 @@ func (vs *version) proxy(r *http.Request, peers []string) ([]byte, error) {
 	for {
 		stageTimeout := time.NewTimer(vs.sequins.config.Sharding.ProxyStageTimeout.Duration)
 		if peerIndex < len(peers) {
-			url := fmt.Sprintf("http://%s%s?proxy=%s", peers[peerIndex], r.URL.Path, vs.name)
-			go vs.proxyAttempt(url, responses, cancel)
+			host := peers[peerIndex]
+			// Should we hit dns?
+			hostName := strings.Split(host, ":")[0]
```

Comment:
there's a `net.SplitHostPort` you can use here. But I'd actually just include the port in the header, since that's the way we refer to the peer everywhere else.